06-building-a-program-start-to-finish/solving-the-problem/output/invertedindexutil.js

```diff
@@ -0,0 +1,254 @@
+          result.push(index);
+        }
+      });
+    });
+    return result;
+  }
+
+  /**
+   * This method remove all the duplicate words in an Array
+   * @param {Array} words - an array of words
+   * @returns {Array} - a new array with no duplicate words
+   */
+  static removeDuplicateWords(words) {
+    const uniqueWords = [];
+    let index = 0;
+    words.forEach((word) => {
+      if (words.lastIndexOf(word) === index) {
+        uniqueWords.push(word);
+      }
+      index += 1;
+    });
+    return uniqueWords;
+  }
+
+   /**
+   * This methods gets all the tokens in a text
+   * @param {String} text - A text of words
+   * @returns {Array} - An array consisting of all the tokens in the text
+   */
+  static tokenize(text) {
+    return text.trim().replace(/-/g, ' ')
+    .replace(/[^A-z\s]/g, '')
+    .toLowerCase()
+    .split(' ');
+  }
+
+  /**
+   * @static
+   * @param {Array} words - An array of all the words in the text of the book
+   * @param {Array} wordsInEachBook - An Array of all the words in each book
+   * @param {Array} bookTitles - An array that consist of a sub array
+   * of all the words in each book
+   * @returns {Array} - An array consisiting of a sub array with each word
+   * and booleans to indicate their occurence in each book
+   * @memberOf InvertedIndexUtility
+   */
+  static constructIndex(words, wordsInEachBook, bookTitles) {
+    const checked = this.checkOccurence(words, wordsInEachBook);
+    const booksLength = bookTitles.length;
+    const booksChecked = this.getEachBookOccurence(checked, booksLength);
+    const indexConstructed = this.mapWordsWithOccurence(words, booksChecked);
+    return indexConstructed;
+  }
+
+  /**
+   * This method gets the title of each book in a file
+   * @static
+   * @param {Array} file - The content of the given file
+   * @returns {Array} - The Title of each book in the file
+   * @memberOf InvertedIndexUtility
+   */
+  static getTitlesOfEachBook(file) {
+    const titleList = [];
+    file.forEach(book => titleList.push(book.title));
+    return titleList;
+  }
+
+  /**
+   * @param {Array} file - contains an array of objects that
+   * contain each book with their title and text.
+   * @returns {Array} - an Array that contains all the words in
+   * the book without duplication
+   */
+  static getAllWords(file) {
+    let words = '';
+
+    file.map(book => (words += `${book.text} `));
+    words = this.tokenize(words).slice(0, words.length - 1);
+    words = this.removeDuplicateWords(words);
+
+    return words.slice(0, words.length);
+  }
+
+  /**
+   * This function returns an array that consists of all the words in each book
+   * [[], [], ]
+   * @static
+   * @param {Array} file - contains an array of objects that
+   * contain each book with their title and text.
+   * @returns {Array} - an array that consist of all the words in each book
+   * @memberOf InvertedIndexUtility
+   */
+  static getAllWordsInEachBook(file) {
+    const doc = [];
```

Comment:
Is "doc" a word or an abbreviation? Can this be written in full such that it communicates correctly?